search/utils.py

```diff
@@ -65,3 +70,15 @@ def upper_string(self):
     def lower_string(self):
         """ use isoformat for _lower date's string format """
         return self._lower.isoformat()
+
+
+def doc_type_warning_decorator(func):
+    """
+    Decorator which mark function with doc_type as deprecated.
+    """
+    @functools.wraps(func)
+    def wrapper(ref, *args, **kwargs):
+        if kwargs.get("doc_type"):
+            log.warning("%s  doc_type parameter is deprecated.", str(func.__code__))
```

Comment:
No logging here. Please raise this exception https://docs.python.org/2/library/exceptions.html#exceptions.DeprecationWarning

Reply:
Thanks